manager/dispatcher/dispatcher.go

```diff
@@ -632,6 +643,11 @@ func (d *Dispatcher) processUpdates(ctx context.Context) {
 					return nil
 				}
 
+				// Update scheduling delay metric for running tasks.
+				if status.State == api.TaskStateRunning {
+					schedulingDelayTimer.Update(time.Duration(status.AppliedAt.Nanos - task.Meta.CreatedAt.Nanos))
```

Comment:
You are only looking at the `Nanos` field, and ignoring the seconds field. If the second changes between these two timestamps, the result won't be accurate. You should use `gogotypes.DurationFromProto` to convert each timestamp to a `time.Time`, and then do normal time math on them.